packages/astro/src/runtime/server/hydration.ts

```diff
@@ -1,12 +1,11 @@
 import type { AstroComponentMetadata, SSRLoadedRenderer } from '../../@types/astro';
 import type { SSRElement, SSRResult } from '../../@types/astro';
 import { hydrationSpecifier, serializeListValue } from './util.js';
-import serializeJavaScript from 'serialize-javascript';
 
 // Serializes props passed into a component so that they can be reused during hydration.
 // The value is any
 export function serializeProps(value: any) {
-	return serializeJavaScript(value);
+	return JSON.stringify(value);
```

Comment:
We already discussed this change, but we need to at least make sure that these props are safe to inject into the `<script>` tags we generate. If one of the props contains a string with `</script>` it will currently break. 

This isn't the cleanest but it probably would work:

```js
JSON.stringify(value).replace(/\<\/script\>/g, '</"+"script>')
```

Reply:
I was able to bring back `serialize-javascript` by aliasing its dependency to the browser-compat version.